packages/flutter/lib/src/material/animation_theme.dart

```diff
@@ -0,0 +1,133 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+
+/// Overrides the default properties values for descendant [Badge] widgets.
+///
+/// Widgets obtain the current [AnimationThemeData] object
+/// using `AnimationTheme.of(context)`. Instances of [AnimationThemeData] can
+/// be customized with [AnimationThemeData.copyWith].
+///
+/// Typically a [AnimationThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.animationTheme].
+///
+/// All [AnimationThemeData] properties have values by default,
+/// and can be changed by using [ThemeData.animationTheme].
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+
+@immutable
+class AnimationThemeData with Diagnosticable {
+  /// Creates the set of color, style, and size properties used to configure [Badge].
+  const AnimationThemeData({
+    this.animationCurve = Curves.easeInOut,
+    this.animationDuration = const Duration(milliseconds: 250),
```

Comment:
themes should have no defaults (that way they aren't breaking)

Reply:
If we remove default values, how is `AnimationTheme.of(context).animationDuration` supposed to work?

Reply:
It would return null by default. You can learn more about this at https://docs.google.com/document/d/1kzIOQN4QYfVsc5lMZgy_A-FWGXBAJBMySGqZqsJytcE/edit#heading=h.iznsi7g4k48o

Reply:
from my understanding, if we were to implement `AnimationTheme` for `AnimatedContainer` the property in framework would be so?
```
duration: AnimationTheme.of(context).duration ?? const Duration(seconds:1)
```

Reply:
Yes, the second part being the current default

Reply:
okay, so unless the user explicitly defines `AnimationTheme`, the widgets will use hardcoded value which is not related to Animation theme. Gotcha!